### PR TITLE
Modifies angular.headroom.js to support updating underlying headroom widget options on directive attribute change

### PR DESCRIPTION
**Pull request by Steake**

### CODE REVIEW

**Comment by atsoy** [comment]
this could be helpful for me too, since my scroller element isn't loaded yet, where headroom was already initialized. 

But i modified it by setting timeout for the entire link function.

```
link: function ($scope, $element) {
        setTimeout(function () {
          var options = {};
          var opts = HeadroomService.options;
          for (var prop in opts) {
            options[prop] = $scope[prop] || opts[prop];
          }
          if ($scope.scroller) {
            options.scroller = document.querySelector($scope.scroller);
          }
          var headroom = new HeadroomService($element[0], options).init();
          $scope.$on('$destroy', function(){
            headroom.destroy();
          });
        }, 1000)
      }
```

File: src/angular.headroom.js [hunk 1]
```diff
@@ -25,6 +25,29 @@
...
+          }
+        });
+        $scope.$watch('classes', function (newClasses) {
+          if (newClasses) {
+            headroom.classes = newClasses;
+          }
+        });
+        $scope.$watch('scroller', function (newScroller) {
+          if (newScroller) {
+            headroom.scroller = newScroller;
```
Comment (atsoy) [inline src/angular.headroom.js]: `newScroller` is just a string (mentioned in conversation comments already).

`headroom.scroller = document.querySelector(newScroller);`

Comment (atsoy) [inline src/angular.headroom.js]: `newScroller` is just a string (mentioned in conversation comments already).

`headroom.scroller = document.querySelector(newScroller);`

Comment (Steake) [inline src/angular.headroom.js]: @atsoy Ah yes. Makes sense to me, lets go with your suggestion.

**Comment by WickyNilliams** [comment]
Thanks for this @Steake. 

My angular skillz are non-existent, so while it looks sensible, I'm unable to review properly. @toddmotto does this look OK to you?

**Comment by toddmotto** [comment]
Haven't tested the code but looks fine - AngularJS has moved on a lot in v1.5 and beyond, things like the `.component()` method introduces lifecycle hooks and one-way dataflow, this could help improve some of the performance side (removing `$watch` etc). Perhaps two versions would be suitable, one for v1.4 and below and v1.5 and above?

**Comment by Steake** [comment]
@toddmotto @WickyNilliams  Shall we merge this with master and then look at updating for v1.5+ ?

**Comment by tabish89** [comment]
@toddmotto @Steake -- I am unable to get this to work with our angular environment.  After doing a little research, I wanted to confirm if this is still a supported plugin (++ for, Angular).  

Site is developed using angular 1.2.x.  Not sure about backwards compatibility (angular-min/css has caused us some trouble by not being fully backward compatible).

Anyway, not sure why I headroom.js is not working.  The following has been done so far, but for the life of me cannot figure out why it is not working or what is wrong with my implementation:

1. Added headroom.js
2. Added angular.headroom.js
3. Added reference to index.html

```
  <script src="/js/libraries/angular.headroom.js"></script>
  <script src="/js/libraries/headroom.js"></script>
```

4. Added requirement to angularJS module

```
angular.module('PSApp', [
  'fcsa-number',
  'headroom',
  'typer',...
])
```

5. Added some basic CSS

```
.headroom {
    will-change: transform;
    transition: transform 200ms linear;
}

.headroom--pinned {
    transform: translateY(0%);
}

.headroom--unpinned {
    transform: translateY(-100%);
}
```

What am I doing wrong?  It just doesn't want to work... activate.  I scroll up/down and nothing happens...

**Comment by Steake** [comment]
Boys... it's been years. I'm closing this PR.

**Pull request closed without merging**